rails_application/app/services/orders/submit_service.rb

```diff
@@ -0,0 +1,42 @@
+module Orders
+  class SubmitService < ApplicationService
+    def initialize(order_id:, customer_id:)
+      @order_id = order_id
+      @customer_id = customer_id
+    end
+
+    def call
+      success = true
+      unavailable_products = []
+
+      event_store
+        .within { submit_order }
+        .subscribe(to: Processes::ReservationProcessFailed) do |event|
+          success = false
+          unavailable_products = Products::Product.where(id: event.data.fetch(:unavailable_products)).pluck(:name)
+        end
+        .call
+
+      if success
+        Result.new(:success)
+      else
+        Result.new(:products_out_of_stock, unavailable_products)
+      end
+      rescue Ordering::Order::IsEmpty
+        Result.new(:order_is_empty)
+      rescue Crm::Customer::NotExists
+        Result.new(:customer_not_exists)
```

Comment:
`call` method returns a `Result` object which represents a logical path of execution combined with optional arguments (e.g. `unavailable_products`).